src/utils/user.authentication.ts

```diff
@@ -0,0 +1,43 @@
+import * as _ from 'lodash';
+import {Credentials, UserRepository} from '../repositories/user.repository';
+import {toJSON} from '@loopback/testlab';
+import {promisify} from 'util';
+import * as isemail from 'isemail';
+import {HttpErrors} from '@loopback/rest';
+const jwt = require('jsonwebtoken');
+const signAsync = promisify(jwt.sign);
+
+export async function getAccessTokenForUser(
+  userRepository: UserRepository,
+  credentials: Credentials,
+): Promise<string> {
+  const foundUser = await userRepository.findOne({
+    where: {email: credentials.email, password: credentials.password},
```

Comment:
Please see the discussion in https://github.com/strongloop/loopback4-example-shopping/pull/26#discussion_r246744678 where I am explaining why storing passwords in plain-text is a security vulnerability.

To me, this is a blocker preventing landing of this pull request.

Reply:
On the second thought, I see that this problem was not introduced by your pull request, it was already there on `master`, so maybe it's alright to leave the fix out of scope of this pull request.

I am proposing the following approach:
- Create a new issue in this repository to keep track of this problem and allow us to plan the work in one of the next milestones.
- Here in this pull request, add a prominent code comment to point out the problem and refer people to the GitHub issue for further discussion.

Reply:
@bajtos Given a look of https://github.com/strongloop/loopback-next/issues/1996 I take back my word :( Let's address it in a separate story.

Reply:
Actually, the scope of strongloop/loopback-next#1996 is different. When I was creating that issue, I did not realize that we have a bigger problem in the current implementation!

As I see it, we have two changes to make:
- MUST HAVE: hash the passwords before they are stored in the database. I think we need to create a new issue for this?
- SHOULD HAVE: move credentials (password, salt?) from User model to a new model. This is covered by  strongloop/loopback-next#1996.

Please don't forget to add a comment to this place in code to warn users about not using this code as a reference implementation.